cg/store/api/find_business_data.py

```diff
@@ -330,13 +330,67 @@ def get_number_of_reads_for_sample_passing_q30_threshold(
         reads_count: Optional[int] = total_reads_query.scalar()
         return reads_count if reads_count else 0
 
+    def get_average_q30_for_sample_on_flow_cell_from_sample_lane_metrics(
+        self, sample_internal_id: str, flow_cell_name: str
+    ) -> float:
+        sample_lanes: List[SampleLaneSequencingMetrics] = apply_metrics_filter(
+            metrics=self._get_query(table=SampleLaneSequencingMetrics),
+            filter_functions=[
+                SequencingMetricsFilter.FILTER_BY_FLOW_CELL_NAME,
+                SequencingMetricsFilter.FILTER_BY_SAMPLE_INTERNAL_ID,
+            ],
+            sample_internal_id=sample_internal_id,
+            flow_cell_name=flow_cell_name,
+        ).all()
+
+        return sum(
+            [sample_lane.sample_base_fraction_passing_q30 for sample_lane in sample_lanes]
+        ) / len(sample_lanes)
+
+    def get_average_fraction_passing_q30_from_sample_lane_metrics(
+        self, flow_cell_name: str
+    ) -> float:
+        """Get average q30 on a given flow cell."""
```

Comment:
More verbose docstring
```suggestion
        """Calculates the average q30 for each sample on a flow cell and returns the average between the samples."""
```